plugin/src/test/java/org/opensearch/ml/task/MLTrainingTaskRunnerTests.java

```diff
@@ -0,0 +1,323 @@
+        verify(mlTaskManager).add(any(MLTask.class));
+        verify(mlTaskManager).remove(anyString());
+        verify(mlIndicesHandler, never()).initModelIndexIfAbsent(any());
+        verify(client, never()).index(any(), any());
+        verify(taskRunner).handleAsyncMLTaskFailure(any(MLTask.class), any(Exception.class));
+    }
+
+    public void testExecuteTask_OnLocalNode_AsyncRequest() {
+        setupMocks(true, false, false, false);
+        taskRunner.executeTask(asyncRequestWithDataFrame, transportService, listener);
+        verify(listener).onResponse(any());
+        verify(mlTaskManager).createMLTask(any(MLTask.class), any());
+        verify(mlTaskManager).add(any(MLTask.class));
+        verify(mlTaskManager).remove(anyString());
+        verify(mlIndicesHandler).initModelIndexIfAbsent(any());
+        verify(client).index(any(), any());
+    }
+
+    public void testExecuteTask_OnLocalNode_AsyncRequest_FailToCreateTask() {
+        setupMocks(true, true, false, false);
+        taskRunner.executeTask(asyncRequestWithDataFrame, transportService, listener);
+        verify(listener, never()).onResponse(any());
+        ArgumentCaptor<Exception> argumentCaptor = ArgumentCaptor.forClass(Exception.class);
+        verify(listener).onFailure(argumentCaptor.capture());
+        assertEquals(errorMessage, argumentCaptor.getValue().getMessage());
+        verify(mlTaskManager).createMLTask(any(MLTask.class), any());
+        verify(mlTaskManager, never()).add(any(MLTask.class));
+        verify(mlTaskManager, never()).remove(anyString());
+        verify(mlIndicesHandler, never()).initModelIndexIfAbsent(any());
+        verify(client, never()).index(any(), any());
+        // don't need to update async task if ML task not created
+        verify(taskRunner, never()).handleAsyncMLTaskFailure(any(MLTask.class), any(Exception.class));
+    }
+
+    public void testExecuteTask_OnLocalNode_AsyncRequest_FailToCreateTaskWithException() {
+        setupMocks(true, true, true, false);
+        taskRunner.executeTask(asyncRequestWithDataFrame, transportService, listener);
+        verify(listener, never()).onResponse(any());
+        ArgumentCaptor<Exception> argumentCaptor = ArgumentCaptor.forClass(Exception.class);
+        verify(listener).onFailure(argumentCaptor.capture());
+        assertEquals(errorMessage, argumentCaptor.getValue().getMessage());
+        verify(mlTaskManager).createMLTask(any(MLTask.class), any());
+        verify(mlTaskManager, never()).add(any(MLTask.class));
+        verify(mlTaskManager, never()).remove(anyString());
+        verify(mlIndicesHandler, never()).initModelIndexIfAbsent(any());
+        verify(client, never()).index(any(), any());
+        // don't need to update async task if ML task not created
+        verify(taskRunner, never()).handleAsyncMLTaskFailure(any(MLTask.class), any(Exception.class));
+    }
+
+    public void testExecuteTask_OnRemoteNode_SyncRequest() {
```

Comment:
Maybe add a new test in this test class to catch the exception like NodeNotConnectedException when transportService.sendRequest?

Reply:
Will add in next PR